src/coreComponents/mesh/MeshBase.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file MeshBase.hpp
+ */
+
+#ifndef GEOS_MESH_MESHBASE_HPP
+#define GEOS_MESH_MESHBASE_HPP
+
+#include "dataRepository/Group.hpp"
+#include "dataRepository/WrapperBase.hpp"
+#include "codingUtilities/Utilities.hpp"
+#include "common/DataTypes.hpp"
+
+
+namespace geos
+{
+
+/**
+ *  @class MeshBase
+ *  @brief The MeshBase class provides an abstract base class implementation for different mesh types.
+ *	   The MeshBase is the Group specialization for different type of mesh handling.
+ */
+class MeshBase : public dataRepository::Group
```

Comment:
So `MeshBase` is currently just to hold the catalog definition/instantiation? Is there plans for specifying a pure virtual interface here?

Reply:
For now no. It is to avoid implementing the API of MeshGeneratorBase